WIP - Add Notebook Authoring Guide (#127)
jupyter/accessibility: docs/resources/notebook-authoring-guide.md

# Notebook Authoring Accessibility Guide

Writing notebooks is an art, depending on what as an author you want to share to the world the way that the writing and publishing techniques can vary widely. There are some writing and publishing techniques that you can use to ensure your notebooks' content is more accessible.

In this section, we will introduce some guidelines to make edits to your notebooks for accessibility. It is an excellent step you can take to make our communities more inclusive!

All the information presented is based on the [draft authoring tips](https://github.com/Iota-School/notebooks-for-all/blob/main/resources/event-hackathon/accessibility-tips-for-jupyter-notebooks.md) and the [Notebook authoring accessibility checklist](https://github.com/Iota-School/notebooks-for-all/blob/main/resources/event-hackathon/notebook-authoring-checklist.md) from the Notebooks For All team.

> # Table of contents

Author: I think it is important to point where all the original information is

1. [Structure](#structure)
2. [Text](#text)
3. [Code](#code)
4. [Media](#media)
    1. [Images](#images)
    2. [Visualizations](#visualizations)
    3. [Videos](#videos)
    4. [Audio and Sonifications](#audio)
    5. [Interactive Widgets](#interactive)
5. [Resources](#resources)
6. [Acknowledgements](#acknowledgements)

## Structure <a name="structure"></a>

Sighted users are able to perceive structure and relationship using different visual cues. We want to ensure that **information and relationships** are perceivable to all. In order to accomplish this goal, we as content authors need to provide access to the information in different modalities.

> We will discuss in this section some recommendations for the most common elements in the structure of your notebooks.

Reviewer: Encouraging people to use content headings since they are currently the main means of structure in notebooks. 

```suggestion
At the time of writing, notebooks gain most of their structure from their content and content headings (like `<h1>`, `<h2>`, etc. or the Markdown equivalents `#`, `##`, etc.). Making sure to use headings and other markdown as intended is one of the most important things an author can do to make an approachable and navigable notebook. 
```

### The First Cell

- [ ] The title of the notebook in a first-level heading (eg. `<h1>` or `# in markdown`).
- [ ] A brief description of the notebook.
- [ ] A brief summary of the notebook.
- [ ] The date first published.
- [ ] The author(s) and affiliation(s) (if relevant).
- [ ] The date last edited (if relevant).
- [ ] A link to the notebook's source(s) (if relevant).
- [ ] A table of contents in an [ordered list](https://www.markdownguide.org/basic-syntax/#ordered-lists) (`1., 2.,` etc. in Markdown).

> ### The Rest of the Cells
> 
> - [ ] There is only one H1 (`#` in Markdown) used in the notebook.
> - [ ] The notebook uses other [heading tags](https://www.markdownguide.org/basic-syntax/#headings) in order without skipping numbers.

Reviewer: Little edits for specificity. This has been awkward to write in every iteration, so I'm always experimenting. Open to suggestion.

```suggestion
### Other cells

- [ ] There should be only one H1 (`#` in Markdown) used in the notebook; it should not appear in other cells.
- [ ] The notebook uses other [heading tags](https://www.markdownguide.org/basic-syntax/#headings) in order throughout. This creates a table of contents that supports the whole notebook.
```

> ## Text <a name="text"></a>
> 
> - [ ] All link text is descriptive. It tells users where they will be taken if they open the link.
> - [ ] Use plain language wherever possible.
> - [ ] All acronyms are defined at least the first time they are used.
> - [ ] Field-specific/specialized terms are used when needed, but not excessively.
> - [ ] Text is broken into paragraphs and/or cells where relevant.
> - [ ] Text is in complete sentences where relevant.
> 
> ## Code <a name="code"></a>

Reviewer: Adding cell types to match how notebook interfaces usually refer to text versus code.
```suggestion
## Text (Markdown cells)<a name="text"></a>

- [ ] All link text is descriptive. It tells users where they will be taken if they open the link.
- [ ] Use plain language wherever possible.
- [ ] All acronyms are defined at least the first time they are used.
- [ ] Field-specific/specialized terms are used when needed, but not excessively.
- [ ] Text is broken into paragraphs and/or cells where relevant.
- [ ] Text is in complete sentences where relevant.

## Code (code cells) <a name="code"></a>
```

## Media <a name="media"></a>

This category include images, visualizations, videos and interactive widgets. If you are reviewing a notebook with mixed content please keep in mind the following two points,

- Text is flexible. Whether it is in the document or linked out, text can be read visually, be read audibly, be magnified, or be translated to another language. Having a text alternative is a good back up plan.

- Having enough color contrast is required on almost all visual content.

### Images <a name="images"></a>

- [ ] All images (jpg, png, svgs) have an [image description](https://www.w3.org/WAI/tutorials/images/decision-tree/). This could be
    - [ ] [Alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt) (an `alt` property)
    - [ ] [Empty alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt#decorative_images) for decorative images/images meant to be skipped (an `alt` attribute with no value)
    - [ ] Captions
    - [ ] If no other options will work, the image is described in surrounding paragraphs.

> - [ ] Any [text present in images](https://www.w3.org/WAI/WCAG21/Understanding/images-of-text.html) exists in a text form outside of the image (this can be alt text, captions, or surrounding text.)

Reviewer: Or maybe you don't want this here since you mention it at `## Media`!
